[date] link/build failure

**Host Environment**
- OS: Windows, Linux, macOS
- Compiler: clang-7, gcc-9, AppleClang 11.0.0 (clang-1100.0.33.8)

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install date`

Add appropriate entries in CMakeLists, e.g.

https://github.com/acgetchell/CDT-plusplus/commit/170f7c36e80daf78ddc89309e90fceb4f470ccb2

**Failure logs** 

Windows:
https://ci.appveyor.com/project/acgetchell/cdt-plusplus/builds/29066845#L1214

Linux:
https://travis-ci.org/acgetchell/CDT-plusplus/jobs/615883747#L3777

macOS:
```
╭─adam@hapkido ~/CDT-plusplus/tools ‹feature/date› 
╰─$ ./build.sh 
-- The CXX compiler identification is AppleClang 11.0.0.11000033
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Boost: /Users/adam/vcpkg/installed/x64-osx/include (found version "1.71.0")  
-- Using header-only CGAL
-- Targetting Unix Makefiles
-- Using /Library/Developer/CommandLineTools/usr/bin/c++ compiler.
-- DARWIN_VERSION=19
-- Mac Leopard detected
-- Found GMP: /Users/adam/vcpkg/installed/x64-osx/debug/lib/libgmp.a  
-- Found MPFR: /Users/adam/vcpkg/installed/x64-osx/debug/lib/libmpfr.a  
-- Found Boost: /Users/adam/vcpkg/installed/x64-osx/include (found suitable version "1.71.0", minimum required is "1.48")  
-- Boost include dirs: /Users/adam/vcpkg/installed/x64-osx/include
-- Boost libraries:    
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/adam/CDT-plusplus/build
Scanning dependencies of target initialize
[  6%] Building CXX object CMakeFiles/initialize.dir/src/initialize.cpp.o
[ 13%] Linking CXX executable initialize
[ 13%] Built target initialize
Scanning dependencies of target CDT_test
[ 20%] Building CXX object tests/CMakeFiles/CDT_test.dir/main.cpp.o
[ 26%] Building CXX object tests/CMakeFiles/CDT_test.dir/Ergodic_moves_3_test.cpp.o
[ 33%] Building CXX object tests/CMakeFiles/CDT_test.dir/Foliated_triangulation_test.cpp.o
[ 40%] Building CXX object tests/CMakeFiles/CDT_test.dir/Geometry_test.cpp.o
[ 46%] Building CXX object tests/CMakeFiles/CDT_test.dir/Manifold_test.cpp.o
[ 53%] Building CXX object tests/CMakeFiles/CDT_test.dir/Move_command_test.cpp.o
[ 60%] Building CXX object tests/CMakeFiles/CDT_test.dir/S3ActionTest.cpp.o
[ 66%] Building CXX object tests/CMakeFiles/CDT_test.dir/SphereTest.cpp.o
[ 73%] Building CXX object tests/CMakeFiles/CDT_test.dir/Tetrahedron_test.cpp.o
[ 80%] Building CXX object tests/CMakeFiles/CDT_test.dir/Torus_test.cpp.o
[ 86%] Building CXX object tests/CMakeFiles/CDT_test.dir/Utilities_test.cpp.o
[ 93%] Building CXX object tests/CMakeFiles/CDT_test.dir/VertexTest.cpp.o
[100%] Linking CXX executable CDT_test
Undefined symbols for architecture x86_64:
  "date::current_zone()", referenced from:
      currentDateTime() in Utilities_test.cpp.o
  "date::time_zone::get_info_impl(std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1l> > >) const", referenced from:
      date::sys_info date::time_zone::get_info<std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000l> > >(std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000l> > >) const in Utilities_test.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [tests/CDT_test] Error 1
make[1]: *** [tests/CMakeFiles/CDT_test.dir/all] Error 2
make: *** [all] Error 2
```

**Additional context**
https://github.com/microsoft/vcpkg/pull/9006

Adding `date::tz` to `target_link_libraries` as suggested by the install yields:
```
"/Users/adam/Library/Application Support/JetBrains/Toolbox/apps/CLion/ch-0/192.7142.39/CLion.app/Contents/bin/cmake/mac/bin/cmake" -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=/Users/adam/vcpkg/scripts/buildsystems/vcpkg.cmake -G "CodeBlocks - Unix Makefiles" /Users/adam/CDT-plusplus
-- Found Boost: /Users/adam/vcpkg/installed/x64-osx/include (found version "1.71.0")  
-- Using header-only CGAL
-- Targetting Unix Makefiles
-- Using /Library/Developer/CommandLineTools/usr/bin/c++ compiler.
-- DARWIN_VERSION=19
-- Mac Leopard detected
-- Found Boost: /Users/adam/vcpkg/installed/x64-osx/include (found suitable version "1.71.0", minimum required is "1.48")  
-- Boost include dirs: /Users/adam/vcpkg/installed/x64-osx/include
-- Boost libraries:    
-- Configuring done
CMake Error at /Users/adam/vcpkg/scripts/buildsystems/vcpkg.cmake:166 (_add_executable):
  Target "initialize" links to target "Threads::Threads" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:137 (add_executable)
```

--- a/ports/date/0002-fix-cmake-3.14.patch
+++ b/ports/date/0002-fix-cmake-3.14.patch
@@ -1,24 +1,25 @@
-diff --git i/CMakeLists.txt w/CMakeLists.txt
-index 885e424..b9e0b43 100644
---- i/CMakeLists.txt
-+++ w/CMakeLists.txt
-@@ -72,7 +72,6 @@ target_sources( date INTERFACE
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index ad74900..42d55af 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -72,7 +72,7 @@ target_sources( date INTERFACE
+     $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include/date/iso_week.h>
      $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include/date/julian.h>
  )
- # public headers will get installed:
--set_target_properties( date PROPERTIES PUBLIC_HEADER include/date/date.h )
- target_compile_definitions( date INTERFACE
-     #To workaround libstdc++ issue https://github.com/HowardHinnant/date/issues/388
-     ONLY_C_LOCALE=$<IF:$<BOOL:${COMPILE_WITH_C_LOCALE}>,1,0>
-@@ -112,7 +111,6 @@ if( BUILD_TZ_LIB )
+-if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.15)
++if (0)
+     # public headers will get installed:
+     set_target_properties( date PROPERTIES PUBLIC_HEADER include/date/date.h )
+ endif ()
+@@ -115,7 +115,6 @@ if( BUILD_TZ_LIB )
      endif( )
-     set_target_properties( tz PROPERTIES
+     set_target_properties( date-tz PROPERTIES
          POSITION_INDEPENDENT_CODE ON
 -        PUBLIC_HEADER "${TZ_HEADERS}"
          VERSION "${PROJECT_VERSION}"
          SOVERSION "${PROJECT_VERSION}" )
      if( NOT MSVC )
-@@ -136,8 +134,8 @@ write_basic_package_version_file( "${version_config}"
+@@ -140,8 +139,8 @@ write_basic_package_version_file( "${version_config}"
      COMPATIBILITY SameMajorVersion )
  
  install( TARGETS date
@@ -26,6 +27,6 @@ index 885e424..b9e0b43 100644
 -    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/date )
 +    EXPORT dateConfig )
 +install( FILES include/date/date.h ${TZ_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/date )
- export( TARGETS date NAMESPACE date:: FILE dateConfig.cmake )
- 
- if( BUILD_TZ_LIB )
+ export( TARGETS date NAMESPACE date:: FILE dateTargets.cmake )
+ if (CMAKE_VERSION VERSION_LESS 3.15)
+     install(

--- a/ports/date/CONTROL
+++ b/ports/date/CONTROL
@@ -1,5 +1,5 @@
 Source: date
-Version: 2019-11-08
+Version: 3.0.0
 Homepage: https://github.com/HowardHinnant/date
 Description: A date and time library based on the C++17 <chrono> header
 

--- a/ports/date/portfile.cmake
+++ b/ports/date/portfile.cmake
@@ -1,6 +1,4 @@
-include(vcpkg_common_functions)
-
-if(NOT VCPKG_CMAKE_SYSTEM_NAME OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
+if(VCPKG_TARGET_IS_WINDOWS)
   message(WARNING
     "You will need to also install https://raw.githubusercontent.com/unicode-org/cldr/master/common/supplemental/windowsZones.xml into your install location.\n"
     "See https://howardhinnant.github.io/date/tz.html"
@@ -10,25 +8,24 @@ endif()
 vcpkg_from_github(
   OUT_SOURCE_PATH SOURCE_PATH
   REPO HowardHinnant/date
-  REF 3e376be2e9b4d32c946bd83c22601e4b7a1ce421
-  SHA512 9dad181f8544bfcff8c42200552b6673e537c53b34fbad11663d6435d4e5fd5a3ac6cabbb76312481c9784b237151d9ccd161bb1b8c54c563fa75073896f3cff
+  REF cac99da8dc88be719a728dc1b597b0ac307c1800 #3.0.0
+  SHA512 07bac40c9d92ed92f05ab71b07c203fc341cd35999f1eab16d584bf77ff69e2cdc106931b2faf0dcfc5a311ee55e8445a81fd97c62f4672957b6aac1b24a08fd
   HEAD_REF master
   PATCHES
-    "${CMAKE_CURRENT_LIST_DIR}/0001-fix-uwp.patch"
-    "${CMAKE_CURRENT_LIST_DIR}/0002-fix-cmake-3.14.patch"
+    0001-fix-uwp.patch
+    0002-fix-cmake-3.14.patch
 )
-
-set(DATE_USE_SYSTEM_TZ_DB 1)
-if("remote-api" IN_LIST FEATURES)
-  set(DATE_USE_SYSTEM_TZ_DB 0)
-endif()
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    INVERTED_FEATURES
+    remote-api USE_SYSTEM_TZ_DB
+)
 
 vcpkg_configure_cmake(
   SOURCE_PATH ${SOURCE_PATH}
   PREFER_NINJA
   OPTIONS
+     ${FEATURE_OPTIONS}
     -DBUILD_TZ_LIB=ON
-    -DUSE_SYSTEM_TZ_DB=${DATE_USE_SYSTEM_TZ_DB}
 )
 
 vcpkg_install_cmake()

